### ISSUE TITLE
Python build fails on master

### ISSUE THREAD

**Issue by tonydifranco** [facebookresearch/fastText#678]
Result from the current master branch (41a0f39) with `git clone` and `pip install`
```
      *__result = __unary_op(*__first);
                                     ^
    python/fastText/pybind/fasttext_pybind.cc:275:20: note: candidate is:
                     [&d](const auto& prediction) {
                        ^
    python/fastText/pybind/fasttext_pybind.cc:275:44: note: pybind11_init_fasttext_pybind(pybind11::module&)::__lambda81::__lambda82
                     [&d](const auto& prediction) {
                                                ^
    python/fastText/pybind/fasttext_pybind.cc:275:44: note:   candidate expects 0 arguments, 1 provided
    error: command 'gcc' failed with exit status 1

    ----------------------------------------
```

I checked out a previous commit (8e68462) and it built without any issues.

Platform info:
OS: `CentOS Linux release 7.3.1611 (Core)`
python: `Python 3.7.1`
pip: `pip 18.1`

**Comment by Celebio** [comment]
Hi @tonydifranco ,
Thank you for reporting the issue.
Which version of gcc do you use?

Regards,
Onur

**Comment by tonydifranco** [comment]
@Celebio  `gcc 4.8.5` currently... thx

**Comment by lrjocean** [comment]
same problem

**Comment by Celebio** [comment]
Hi,
Thank you for reporting the issue.
It looks like a compiler/C++ version issue. We will make sure the code works on most c++11 compilers and we will add more tests while building to fail gently if the compiler is not supported.

Thanks!
Onur

**Comment by tonydifranco** [comment]
just fyi, I have been pip installing from the master branch with the same gcc version for nearly a year before it started failing... I don't know exactly which of the last few commits broke backwards compatibility but it was definitely after 8e68462

**Comment by Celebio** [comment]
Hi all,
That should be fixed by ead79114b017083d56b8f6b54b9092f6125642f5.

Thanks again!
Onur
